server.py

```diff
@@ -148,6 +148,46 @@ def getUser():
 	return jsonify(info), 200
 
 
+@app.route('/update/password', methods=['PUT'])
+def changePassword():
+	csrf.protect()
+
+	login_code = session.get('login', None)
+	if login_code is None:
+		return forbidden()
+
+	login = LoginCode.get_by_code(login_code)
+	if login is None:
+		return forbidden()
+
+	json = request.json
+	if json is None:
+		return 'Missing json data', 400
+
+	old = json.get('pass_old', None)
+	new1 = json.get('pass_new', None)
+	new2 = json.get('pass_new_conf', None)
+
+	if old is None or new1 is None or new2 is None:
+		return 'Missing fields', 400
+
+	if not util.isValidPassword(old) or not util.isValidPassword(new1):
+		return 'Invalid password', 400
```

Comment:
We shouldn't check if the old password is a valid password here, the rules might have changed by the time the user has changed their password. This could end up locking the users into not being able to change their password because the old one was invalid.

Reply:
We should only check for equality to the original password.

Reply:
I'm planning on separating valid passwords from acceptable passwords. As in, a None, empty, or non-string password is not valid. The length and/or complexity would be a separate check, likely another utility function called `isAcceptablePassword`.

Reply:
The checks you are doing in isValidPassword actually would be good to also do on usernames then. As it seems more like an isValidUserString function.

Reply:
That's a valid point. We don't currently pass the username around for anything, but when we do get to that functionality we can revisit this utility function and potentially rename it. For now I want to keep it as-is.